internal_test/mySQL_test/database_test.go

```diff
@@ -1,42 +1,169 @@
 	"github.com/matcha-devs/matcha/internal/mySQL"
 )
 
-var db = mySQL.Open()
+// This a variable to hold the test database connection,
+// and run all functions relating to the database:
+var test_db_connection *sql.DB
 
-// TODO(@SeoYoung Cho):what do is the panic error you are having?:
+// Mock environment variable for testing
+// Helper function to create a test database connection and ensure it's clean
+func setupTestDB(t *testing.T) *mySQL.Database {
+	// Get the password from the environment, currently using the admin password this may change in the future!!!
+	password := os.Getenv("MYSQL_PASSWORD")
+	// Connect to MySQL without specifying a database
+	var err error
+	test_db_connection, err = sql.Open("mysql", "root:"+password+"@tcp(localhost:3306)/")
+	if err != nil {
+		t.Fatal("Failed to open test database connection:", err)
+	}
+	if err := test_db_connection.Ping(); err != nil {
+		t.Fatal("Failed to ping test database connection:", err)
+	}
 
-// TODO(@Alishah634): Move Tests to Test Modules and activate modules
+	// Ensure a clean test database by dropping any previous instances
+	_, err = test_db_connection.Exec("DROP DATABASE IF EXISTS test_db")
+	if err != nil {
+		t.Fatal("Failed to drop test database:", err)
+	}
+	// Create a new test database
+	_, err = test_db_connection.Exec("CREATE DATABASE test_db")
+	if err != nil {
+		t.Fatal("Failed to create test database:", err)
+	}
 
-// TODO(@Alishah634): Fix AuthenticateLogin panic error
-func TestAuthenticateLogin(t *testing.T) {
-	t.Log("TestAuthenticateLogin Started")
+	// Close the initial connection and reopen with the new database
+	if err := test_db_connection.Close(); err != nil {
+		log.Fatal("Error closing initial database connection:", err)
+	}
 
-	if db.AuthenticateLogin("Jane", "pw2") != nil {
-		t.Error("Authenticated with incorrect username")
+	// Connect to the newly created test database
+	test_db_connection, err = sql.Open("mysql", "root:"+password+"@tcp(localhost:3306)/test_db")
+	if err != nil {
+		t.Fatal("Failed to open connection to test database:", err)
 	}
-	if db.AuthenticateLogin("Bob", "pw2") == nil {
-		t.Error("Authenticated with incorrect password")
+	if err := test_db_connection.Ping(); err != nil {
+		t.Fatal("Failed to ping test database connection:", err)
 	}
-	if db.AuthenticateLogin("Charlie", "pw4") != nil {
-		t.Error("Authentication failed with correct password")
+
+	// Open the custom testing database with initial SQL setup
+	test_db := mySQL.Open("test_db", "root", password, "../../internal/mySQL/queries/")
+	if test_db == nil {
+		t.Fatal("Failed to open test database")
 	}
-}
 
-// TODO(@Alishah634): implement following API tests
-//TestAddUser
-//
-//TestFindID
-//
-//TestDeleteUser
+	return test_db
+}
 
-func _() {
-	err := db.AddUser("clo", "cotera_hh@gmail.com", "MEXICAN")
+// TestOpenAndClose tests the Open and Close functions
+func TestOpenAndClose(t *testing.T) {
+	test_db := setupTestDB(t)
+	if test_db == nil {
```

Comment:
Shouldn't we be counting to make sure all the necessary tables got generated here? Let's do this later tho bc I have a big change coming up.

Reply:
Noted.